### DISCUSSION TITLE
Voltage divider and ADC. No more burning board.

### DISCUSSION THREAD

**Discussion by IHOXOHI** [micropython/micropython#11953] [General]
Hi,
A tried to check voltage of solar battery... So, from 10-15 Volt... 
I have used a divider with R1=1M ohms, and R2=200K ohms. On a pybstick. At 5V it's work fine... At 13V, I checked with a multimeter without board: 1.9 V, 10 uA on the adc wire.
The voltage regulator has burned when I have connected the board.
... It's not the first one...
I think that my gnd point is shorted by my ADC wire...? So, I need to use R2 less bigger...?
I don't want to Burn other board...
Thanks.

**Comment by rkompass** [comment]
In principle such a voltage divider should work. 200K/ (1M+200K) = 1/6. 1/6 * 15V = 2.5V.
You have to consider that connections may not always functional, especially if you try out things e.g. on a breadboard.
To increase safety you might solder (a few diodes connected in series or a zener diode) parallel to the 200K resistor. So that they limit  the voltage you attach to the adc, even if your external voltage is exceeded.

I have no clear idea why your voltage regulator has burned. Modern regulators limit the current even in case of short circuit.
Perhaps the quality or performance of the voltage regulator was not good enough.
I just found that the Me6215c33 that apparently is used is a <= 1ct piece from China. Found no datasheet. You could try to replace it with something you know the data of.
Did you connect the Vin of the regulator to the solar battery? That might have been too much. 50mA * (15V-3V) = 600mW.
The tiny regulator may not be able to dissipate this power without cooling.

**Comment by robert-hh** [comment]
> I don't want to Burn other board...

On many boards the 3.3V regulator is meant to have a 5V input. These regulator have a maximum input voltage of 6-7V. Applying 10 or 15 V breaks them. In order to drive your board from the solar panel, you need a second external regulator to drop the Solar Panel voltage down to 5V.

**Comment by IHOXOHI** [comment]
Hi,
Thanks a lot...
The pybstick, like pyboard, could be powered on vin up to 18V. So, yes, I have connected the board directly to 13v. Without an adc connection, there isn't problem... I thought that the current depends of consumption of the board,..., I could add diode, but the object is to check the voltage... I don't know if the consumption of a diode is known... If it depends of the voltage, I am fine...
I will try with another regulator for powered the board... And report...
Thanks...

**Comment by IHOXOHI** [comment]
With another regulator it works...
I don't understand what happens...?
I repeat, without adc , the board works fine at 13V on the vin...?...

**Comment by robert-hh** [comment]
Can you show the wiring that you made?

**Comment by IHOXOHI** [comment]
![16890732299476866671110999877190](https://github.com/micropython/micropython/assets/63340434/1a691a95-1a67-4ede-97a4-c8aec45c838b)

**Comment by robert-hh** [comment]
Thanks. It's hard to understand why this circuit would break the regulator. With R1 = 1 MOhm, connected to the ADC input, the worst that should happen is a damage to the ADC input. But not even that is to be expected, since the inrush current of about 10 µA would flow through the ESD diode of the Pin into the 3V3 supply. But that would only happen if the R2 connection path is not stable. There have been regulators that break if fed through the output. The data sheet does not mention anything about it. Maybe something else on the board failed and there is a short now at the 3v3 supply.

**Comment by rkompass** [comment]
Did you measure the voltage on adc pin?
A pre-regulation of voltage to e.g. 6-8 V is strongly recommended anyway, because the small original regulator has no heat spreader. To that external regulator it should be easy to attach such a head spreader (some strip of metal should be sufficient). 
Are your connections reliable or are you using a cheap breadboard?

**Comment by IHOXOHI** [comment]
Thanks a lot for yours helpful...
I can check voltage of my solar battery.
I have used a pybstick... I think it's the same for the pyboard... I will try and report if necessary...

**Comment by IHOXOHI** [comment]
Sorry rKompass,
I have used a cheap breadboard...

**Comment by IHOXOHI** [comment]
![16891469682868778650224342542385](https://github.com/micropython/micropython/assets/63340434/75f22374-fac4-4f2e-8b8c-f0f1cd9c2248)

**Reply by robert-hh** [reply]
That wiring is indeed calling for trouble. About breadboard: I still do not know a brand name for good breadboards. Recently I dumped a few freshly purchased breadboards because of their poor quality - bad contacts, almost impossible to insert components. They were cheap in every aspect.

**Comment by IHOXOHI** [comment]
![IMG_20230712_095145](https://github.com/micropython/micropython/assets/63340434/2747d6a9-2b91-4000-91ff-83922b922407)
The dead board...

**Reply by robert-hh** [reply]
The area around the regulator at the Vin pin looks pretty burnt. That requires some amount of energy, way more than can flow through a 1M resistor. I cannot tell from the picture whether it is the Vin Pin or a component next to it, like the capacitor, which created that smoke.

**Reply by rkompass** [reply]
It's probably the voltage regulator: The calculation 50mA * (15V-3V) = 600mW may be adapted to the real valules:
Perhaps 40mA * (12V-3V) = 360mW. As we see from the other picture the pcb is placed so on the black mini-breadboard that the USB plug and the yellow rows of connectors prevent circulation of the hot air around the regulator. I became so hot that its plastic case got molten?

What puzzles me is that you observed the burn at the time you used the ADC. Was there perhaps a coincidence that you used the black breadboard in this case (ADC reading) for displaying the voltage values on the LCD?
So it could be the following causal chain:
ADC reading -> LCD use -> heat shielding of voltage regulator with black mini breadboard -> voltage regulator overheating?

**Comment by IHOXOHI** [comment]
With a pyblite, everything is fine at 13v...without second regulator...
![IMG_20230713_125730](https://github.com/micropython/micropython/assets/63340434/bf7fde96-078f-4f2a-a243-c71f0460819b)

**Reply by rkompass** [reply]
The air may circulate better around the regulator, one might argue here, so better cooling is happening..
If you enclose the regulator in a little box like before with the USB plug case and the two rows of yellow pin holders and a plastic ceiling 1.5 mm above it, it should burn as well (I wouldn't try that). What I want to make a point about is: The power an electric element (resistor, semiconductor or device) generates is determined by the circuitry it is placed in. But this power has to be dissipated away - by air convection. The smaller the element is i.e. the smaller it's surface, the higher it's temperature will become until an equilibrium is reached generated power = dissipated power. If the dissipation is hindered by limited air convection also the temperature will increase until the equilibrium is reached.
Other way round: If the ADC is done with the regulator being free on the upper side of the pybstick (no limited air convection, pybstick turned around so that its pins are upwards, no enclosure on top of it) it (according to this hypothesis) should not burn.

**Comment by IHOXOHI** [comment]
Thanks a lot for your good view.
Effectively, I have tried with others pybsticks, and when the regulator is enclosed, it burn slowly,  nicely. Outdoor, open to air, it's fine.
Thanks.

**Comment by IHOXOHI** [comment]
![IMG_20230714_112141](https://github.com/micropython/micropython/assets/63340434/64f6894b-7088-4dfc-bff1-ccfd1ddf025c)
